### PR TITLE
[2.19.x] G-10041 Add support for old CQL result filters

### PR DESCRIPTION
**Pull request by jrnorth**
#### What does this PR do?
The User.Preferences model will now convert old-style CQL result filters to filter trees on the fly.

#### Who is reviewing it? 
@millerw8
@kcwire 
@shaundmorris 

#### How should this be tested?
1. Open the Solr Admin UI and query the `preferences` core.
2. Copy the `preferences_json_bin` for your user.
3. Base64 decode that string to a JSON file and create a `resultFilter` key with a CQL string value if no such key exists, or replace the existing value with a CQL string value.
a. The CQL string can be as simple as `(\"anyText\" ILIKE '')`, or it can be a more complex query.
4. Base64 encode that JSON file and update your user preferences via the `Documents` page in the Solr Admin UI, filling in the ID of your preferences and your base64-encoded JSON:
Screen Shot 2022-02-16 at 5 59 50 PM
5. Open Intrigue, click on your result filter, and verify that the CQL you added in step 3 was parsed correctly into the result filter.
6. Run some queries and verify your results appear to be filtered correctly by the result filter.

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/User.spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/**
```
Comment (shaundmorris) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/User.spec.js:1]: Newb question... why did you need to add this file?

**Review by shaundmorris — APPROVED** [review]

**Review by kcwire — APPROVED** [review]

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/User.spec.js (hunk 1 above)
Comment (jrnorth) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/User.spec.js:1]: Didn't have to, just wanted to add some unit tests for the fix.

**Comment by jrnorth** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/User.spec.js (hunk 1 above)
Comment (shaundmorris) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/User.spec.js:1]: gotcha, thanks 👍

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js [hunk 2]
```diff
@@ -207,9 +207,14 @@ User.Preferences = Backbone.AssociatedModel.extend({
     this.listenTo(this, 'change:mapHome', this.savePreferences)
   },
   get(attr) {
-    const value = Backbone.AssociatedModel.prototype.get.call(this, attr)
-    if (attr === 'resultFilter' && value && value.type === 'NOT') {
-      return cql.simplify(value)
+    let value = Backbone.AssociatedModel.prototype.get.call(this, attr)
+    if (attr === 'resultFilter') {
+      if (typeof value === 'string') {
```
Comment (millerw8) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js:212]: So, if I am following correctly, the thought is, if we get here.... we have a resultFilter of type string, from a previous version... otherwise it should be an CQL object and we revert to the previous logic?

**Review by millerw8 — APPROVED** [review]
To mimic the. inspiration of this change... one can use a DDF version from 2.15.x, create an empty result filter (anyText contains \<blank string\>) and save it.  Then either
- export out their preferences, stand up a DDF 2.19.x and import those preferences , or 
- stand up a DDF 2.19.x which hits the same solr database

Previous behavior of these steps was, when running a query in the UI that has results, all results would be hidden with an error in the console.  Ideally, with this change, the filter would be respected (in this case, with no results filtered, and so all results displayed), and no errors in console.

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js (hunk 2 above)
Comment (jrnorth) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js:212]: Yes.

**Comment by maiiiiiiii** [comment]
hero successful! 

updating_preferences_solr
ui
